### PR TITLE
Support X-Sendfile in response to a POST request

### PR DESCRIPTION
**Pull request by fgoepel**
I didn't manage to push to the existing branch and deleting it automatically closed the pull request so I'm opening a new PR with the requested changes.

### CODE REVIEW

**Review by boazsegev — CHANGES_REQUESTED** [review]
Perfect, almost done :)

File: ext/iodine/http_internal.c [hunk 1]
```diff
@@ -42,7 +42,8 @@ void http_on_request_handler______internal(http_s *h,
           fiobj_hash_get2(h->headers, fiobj_obj2hash(HTTP_HEADER_ACCEPT)),
           HTTP_HVALUE_SSE_MIME))
     goto eventsource;
-  if (settings->public_folder) {
+  if (settings->public_folder &&
+      (fiobj_obj2cstr(h->method).len != 4 || !strncasecmp("post", fiobj_obj2cstr(h->method).data, 4))) {
```
[the remark this answers is not in the document]
Comment (fgoepel) [reply ext/iodine/http_internal.c]: I think you're right. I've made the change and pushed again.

Another alternative might be to ` !strncasecmp("head", ...`, as that seems to be the only case we want to accept here, that's not already rejected by `len != 4`. But in the end that would only make a difference if a new HTTP method with 4 letters were to get added later.

**Review by boazsegev — APPROVED** [review]

**Comment by boazsegev** [comment]
Perfect, thanks!

**Comment by boazsegev** [comment]
Done. I released the patched version. Enjoy :)

**Comment by fgoepel** [comment]
Thanks.

**Pull request merged**